encodings/dict/src/compute/min_max.rs

```
@@ -1,16 +1,148 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: Copyright the Vortex contributors
 
+use itertools::Itertools;
+use num_traits::Zero;
 use vortex_array::compute::{MinMaxKernel, MinMaxKernelAdapter, MinMaxResult, min_max, take};
-use vortex_array::register_kernel;
+use vortex_array::{Array as _, ToCanonical, register_kernel};
+use vortex_dtype::match_each_unsigned_integer_ptype;
 use vortex_error::VortexResult;
+use vortex_mask::Mask;
 
 use crate::{DictArray, DictVTable};
 
 impl MinMaxKernel for DictVTable {
     fn min_max(&self, array: &DictArray) -> VortexResult<Option<MinMaxResult>> {
-        min_max(&take(array.values(), array.codes())?)
+        let Mask::AllTrue(_) = array.codes().validity_mask() else {
```

can you explain (in a comment) why we only handle all_true specially.

---

I think if any of the codes are null, that's pretty much undefined behavior so I preferred to just fall back to take rather than fiddle with validity masks. Will add a comment.